tests/infrastructure/test_open_search_stack.py

```diff
@@ -0,0 +1,282 @@
+
+# Installed
+from aws_cdk.assertions import Match, Template
+
+# Local
+from sds_data_manager.stacks.opensearch_stack import OpenSearch
+
+
+@pytest.fixture(scope="module")
+def template(app, sds_id, env):
+    stack_name = f"opensearch-{sds_id}"
+    stack = OpenSearch(app, stack_name, sds_id, env=env)
+    template = Template.from_stack(stack)
+    return template
+
+
+def test_secrets_manager_resource_count(template):
+    template.resource_count_is("AWS::SecretsManager::Secret", 1)
+
+
+def test_secrets_manager_resource_properties(template):
+    template.has_resource(
+        "AWS::SecretsManager::Secret",
+        {
+            "DeletionPolicy": "Delete",
+            "UpdateReplacePolicy": "Delete",
+        },
+    )
+
+
+def test_iam_roles_resource_count(template):
+    template.resource_count_is("AWS::IAM::Role", 1)
+
+
+def test_expected_properties_for_iam_roles(template):
+    found_resources = template.find_resources(
+        "AWS::IAM::Role",
+        {
+            "Properties": {
+                "AssumeRolePolicyDocument": {
+                    "Statement": [
+                        {
+                            "Action": "sts:AssumeRole",
+                            "Effect": "Allow",
+                            "Principal": {"Service": "lambda.amazonaws.com"},
+                        }
+                    ],
+                    "Version": "2012-10-17",
+                }
+            }
+        },
+    )
+
+    # There is 1 IAM Role expected resources with the same properties
+    # confirm that all are found in the stack
+    assert len(found_resources) == 1
+
+
+def test_opensearch_domain_resource_count(template):
+    template.resource_count_is("AWS::OpenSearchService::Domain", 1)
+
+
+def test_opensearch_domain_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::OpenSearchService::Domain",
+        {
+            "DomainName": f"sdsmetadatadomain-{sds_id}",
+            "EngineVersion": "OpenSearch_1.3",
+            "ClusterConfig": {"InstanceType": "t3.small.search", "InstanceCount": 1},
+            "EBSOptions": {"EBSEnabled": True, "VolumeSize": 10, "VolumeType": "gp2"},
+            "NodeToNodeEncryptionOptions": {"Enabled": True},
+            "EncryptionAtRestOptions": {"Enabled": True},
+        },
+    )
+
+
+def test_custom_cloudwatch_log_resource_policy_count(template):
+    template.resource_count_is("Custom::CloudwatchLogResourcePolicy", 1)
+
+
+def test_log_groups_resource_count(template):
+    template.resource_count_is("AWS::Logs::LogGroup", 3)
+
+
+def test_sdsmetadatadomain_slow_search_logs_resource_properties(template):
+    template.has_resource(
+        "AWS::Logs::LogGroup",
+        {
+            "DeletionPolicy": "Retain",
+            "UpdateReplacePolicy": "Retain",
+        },
+    )
+    template.has_resource_properties("AWS::Logs::LogGroup", {"RetentionInDays": 30})
+
+
+def test_sdsmetadatadomain_app_logs_resource_properties(template):
```

Comment:
This test is identical to the above. I think this is where you could use the `find_resources()` and count if you wanted to say there should be exactly 3 resources found with these same exact properties. Otherwise, this is not really testing what we think it is here because if one of them changes, this will still pass because the other one will be found.